plugins/tasks/ansible/src/main/java/com/walmartlabs/concord/plugins/ansible/AnsibleInventory.java

```diff
@@ -25,23 +25,94 @@
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
-import java.io.BufferedWriter;
-import java.io.IOException;
+import java.io.*;
 import java.nio.file.Files;
 import java.nio.file.Path;
 import java.nio.file.StandardOpenOption;
 import java.util.*;
+import java.util.concurrent.*;
 import java.util.stream.Collectors;
 
+import static com.walmartlabs.concord.common.LogUtils.withMdc;
+
 public class AnsibleInventory {
 
-    public static void process(AnsibleContext context, PlaybookScriptBuilder playbook) throws IOException {
+    private static final int MAX_HOSTS_VERBOSE = 50;
+
+    public static void process(AnsibleContext context, PlaybookScriptBuilder playbook) throws AnsibleInventoryException, IOException {
         List<String> inventories = new AnsibleInventory(context.workDir(), context.tmpDir(), context.debug())
                 .write(context.args());
 
+        int verbose = MapUtils.getInt(context.args(), "verbose", 0);
+        if (verbose > 0) { // disallow verbose logging with large inventories
+            int invSize = inventories.stream()
+                    .map(i -> countHosts(i, context.workDir()))
+                    .reduce(Integer::sum).orElse(0);
+
+            if (invSize > MAX_HOSTS_VERBOSE) {
+                throw new IllegalArgumentException("Inventory contains " + invSize +
+                        " hosts and verbose logging is enabled. Verbose logging" +
+                        " is not allowed for inventories with greater than " +
+                        MAX_HOSTS_VERBOSE + " hosts.");
+            }
+        }
+
         playbook.withInventories(inventories);
     }
 
+    private static int countHosts(String inventory, Path workDir) {
+        Path inventoryPath = workDir.resolve(inventory);
+        if (!Files.exists(inventoryPath)) {
+            throw new IllegalArgumentException("Cannot find inventory: " + inventoryPath);
+        }
+
+        ProcessBuilder pb = new ProcessBuilder()
+                .command("ansible-inventory", "-i", inventoryPath.toString(), "--list")
+                .directory(workDir.toFile());
+
+        final ExecutorService executor = Executors.newCachedThreadPool();
+
+        try {
+            Process p = pb.start();
+
+            Future<Integer> out = executor.submit(() -> {
+                try (InputStream is = p.getInputStream()) {
+                    // parse ansible-inventory JSON output
```

Comment:
Should `is` just be passed to ansible-playbook since it's already merged into json, to avoid subsequent execution in ansible-playbook? Some of the example inventory [scripts](https://github.com/ansible-collections/community.general/tree/main/scripts/inventory) and [plugins](https://docs.ansible.com/ansible/latest/plugins/inventory.html#inventory-plugins) take a while depending on the data source.